### ISSUE TITLE
ExtractExcel should be pure XML→xlsx (no loader, no RuleSet)

### ISSUE THREAD

**Issue by PaulSnow** [DTRules/DTRules#545]
## Why

The current `excel.ExtractExcel` (#543 / #544) routes every `_dt.xml` and `_edd.xml` through `session.NewRuleSet` + `loader.Load`, which compiles EL expressions and wants to resolve entity references. That is operational machinery; extraction is a **pure representation conversion**.

Extract is:

```
XML (data model) → xlsx (data model)
```

Not:

```
XML → compile → RuleSet → export
```

Compilation / session / entity resolution are only needed if we were going to **update** the decision tables and reinsert them into the binary. That is out of scope for the extract use case.

## Symptoms of the over-engineering

- Spins up 114 throwaway `RuleSet`s for a 114-file TaxReturn tree.
- Emits spurious loader warnings when a DT file references entities defined in a sibling EDD file (because each file gets a fresh empty session).
- Couples extract to the loader, so any loader bug (e.g., #541) leaks into the extract path.

None of these affect correctness of the extracted xlsx — just noise, waste, and unwanted coupling.

## Correct shape

1. XML-unmarshal into the existing data structs in `pkg/dtrules/excel/`:
   - `DecisionTablesXML` (from `dt_importer.go` or wherever it's defined)
   - `EDDXML` + `EDDXMLEntity`
   - `MapXML` (already pure)
2. Feed each model to an exporter path that writes xlsx directly from the model — no `RuleSet`, no `session`.
3. EL text in `*_dsl` fields copies through verbatim. Postfix cells in the XML are ignored — Excel doesn't carry postfix.

`MapExporter.ExportToFile(m *MapXML, path string)` already has this shape. DT and EDD need matching entry points:

```go
func (e *Exporter) WriteDTFromXML(dt *DecisionTablesXML, out string) error
func (e *Exporter) WriteEDDFromXML(edd *EDDXML, out string) error
```

(Or free functions — whichever fits existing API style.)

Then rewrite `ExtractExcel` to:

```go
func ExtractExcel(src fs.FS, dst string) error {
    return fs.WalkDir(src, ".", func(path string, d fs.DirEntry, err error) error {
        // ... read bytes, unmarshal, export, done.
        // No session. No loader. No RuleSet.
    })
}
```

## Acceptance

- No `session` or `loader` import in `pkg/dtrules/excel/extract.go`.
- Existing `TestExtractExcel_*` tests continue to pass.
- Add a test that asserts no warning is emitted when extracting a tree where DT files reference entities defined in sibling EDD files.
- Code comment at top of `extract.go` updated to remove the "per-file RuleSet" limitation note and point at this issue's closure in the commit.

## Relates to

- #543 (the original implementation) — revised here.
- #541 (fs.FS loader) — becomes orthogonal once extract is decoupled from the loader.
